src/go/rpk/pkg/cli/cluster/logdirs.go

```diff
@@ -196,5 +206,11 @@ where revision is a Redpanda internal concept.
 	cmd.Flags().BoolVar(&sortBySize, "sort-by-size", false, "If true, sort by size")
 	cmd.Flags().StringSliceVar(&topics, "topics", nil, "Specific topics to describe")
 	cmd.Flags().StringVar(&aggregateInto, "aggregate-into", "", "If non-empty, what column to aggregate into starting from the partition column (broker, dir, topic)")
+	cmd.Flags().BoolVarP(&human, "human-readable", "H", false, "Print the logdirs size in a human-readable form")
+
+	cmd.RegisterFlagCompletionFunc("aggregate-into", func(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
+		opts := []string{"broker", "dir", "topic"}
+		return opts, cobra.ShellCompDirectiveNoSpace
```

Comment:
Just tested this locally, it's not working for me. NoSpace is needed in profile set because it's setting a []string, and each key=value is a single string entry and is parsed splitting by =. Here, this is just a flag with one string value. This can be ShellCompDirectiveDefault -- note default is used in areas currently where it's setting a string.

I _think_ this isn't working as well because this isn't returning a suffix of toComplete. In profile / -X, the returned options are filtered by if they have toComplete as a prefix.

Reply:
Weird, I tested before sending the PR and I just tested 1 min ago:

![image](https://github.com/redpanda-data/redpanda/assets/59714880/f9c3a038-decf-4071-9b30-89a30fb96ef3)

Let me chase down what might be going on.

Reply:
Also, see that grafana.go doesn't use the toComplete string at all. I just changed the shellCompDirective to the default. Please test again and thanks :pray:

Reply:
We discovered this is a bash thing and affects other commands currently -- we can have a follow up issue to track this down.